### PR TITLE
GOVUK PaaS Github Actions

### PR DESCRIPTION
**Pull request by cpjmcquillan**
Based on govuk-paas branch.
Spike into the use of Github Actions for CI workflow [TEVA-515](https://dfedigital.atlassian.net/browse/TEVA-515).

**In this PR**
- Execute lint checking on push
- Execute tests on push
- Execute build using buildpacks and deploy to staging GOVUK PaaS space using CF on push

**TODO** [TEVA-611](https://dfedigital.atlassian.net/browse/TEVA-611)
- [ ] Formalise a production ready workflow

### CODE REVIEW

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        run: docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+        
+      - name: Build and push docker image
+        env:
+          DOCKER_IMAGE_NAME: ${{ secrets.DOCKER_IMAGE_NAME }}
+        run: | # check docker permissions and ensure correct image name 
+          docker build -t $DOCKER_IMAGE_NAME .
+          docker push $DOCKER_IMAGE_NAME
+      
+      - name: Install CF CLI
```
Comment (tatyree) [inline .github/workflows/build-docker.yml]: I'm OK with shortening Cloud Foundry in `ENV` variable keys, but let's avoid doing it in descriptive steps, please.  No need to change now, just bear it in mind moving forward.

**Comment by cesidio** [comment]
Specs running in 3m 59s!!! Great work! :boom:

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: CI Spike (Build)
```
[the remark this answers is not in the document]
Comment (cpjmcquillan) [reply .github/workflows/build.yml]: It all happens as part of `cf push` using the `manifest.yml` file

File: Dockerfile [hunk 3]
```diff
@@ -39,6 +39,7 @@ COPY Gemfile $INSTALL_PATH/Gemfile
 COPY Gemfile.lock $INSTALL_PATH/Gemfile.lock
 
 RUN gem install bundler
+RUN echo 'export PATH="/usr/local/bundle/bin:/usr/local/bin:$PATH"' >> $HOME/.bashrc
```
[the remark this answers is not in the document]
Comment (cpjmcquillan) [reply Dockerfile]: @cesidio did this as part of his commit  (I just based my branch on his) so I'm not clear on his reasoning behind this!

**Pull request closed without merging**